wipac_dev_tools/timing_tools.py

```diff
@@ -0,0 +1,64 @@
+"""Utilities for timers, interval trackers, etc."""
+
+import asyncio
+import logging
+import time
+
+
+class IntervalTimer:
+    """A utility class to track time intervals.
+
+    This class allows tracking of elapsed time between actions and provides
+    mechanisms to wait until a specified time interval has passed.
+    """
+
+    def __init__(self, seconds: float, logger: logging.Logger) -> None:
+        self.seconds = seconds
+        self._last_time = time.time()
```

Comment:
Recommend using `time.monotonic()` which takes any clock skew, leap seconds, etc out of the equation.

Reply:
done :heavy_check_mark: